x-pack/plugins/security_solution/public/management/components/artifact_list_page/components/artifact_flyout.tsx

```diff
@@ -0,0 +1,203 @@
+    'xpack.securitySolution.artifactListPage.flyoutCreateSubmitButtonLabel',
+    { defaultMessage: 'Add' }
+  ),
+  flyoutEditSubmitButtonLabel: i18n.translate(
+    'xpack.securitySolution.artifactListPage.flyoutCreateSubmitButtonLabel',
+    { defaultMessage: 'Save' }
+  ),
+  flyoutDowngradedLicenseTitle: i18n.translate(
+    'xpack.securitySolution.artifactListPage.expiredLicenseTitle',
+    {
+      defaultMessage: 'Expired License',
+    }
+  ),
+  flyoutDowngradedLicenseInfo: i18n.translate(
+    'xpack.securitySolution.artifactListPage.flyoutDowngradedLicenseInfo',
+    {
+      defaultMessage:
+        'Your Kibana license has been downgraded. Future policy configurations will now be globally assigned to all policies.',
+    }
+  ),
+});
+
+interface ArtifactFormComponentProps {
+  item: object;
+  mode: 'edit' | 'create';
+  /** signals that the form should be made disabled (ex. during update) */
+  disabled: boolean;
+  /** reports the state of the form data and the current updated item */
+  onChange(formStatus: { isValid: boolean; item: object }): void;
+}
+
+export interface ArtifactFlyoutProps {
+  apiClient: unknown; // ExceptionsListApiClient; // FIXME:PT use api client type
+  FormComponent: React.ComponentType<ArtifactFormComponentProps>;
+  onCancel(): void;
+  onSuccess(): void;
+  /**
+   * If the artifact data is provided and it matches the id in the URL, then it will not be
+   * retrieved again via the API
+   */
+  artifact?: ExceptionListItemSchema;
+  /** Any label overrides */
+  labels?: Partial<typeof ARTIFACT_FLYOUT_LABELS>;
+  showExpiredLicenseBanner?: boolean;
+  'data-test-subj'?: string;
+  size?: EuiFlyoutSize;
+}
+
+/**
+ * Show the flyout based on URL params
+ */
+export const MaybeArtifactFlyout = memo<ArtifactFlyoutProps>(
+  ({
+    FormComponent,
+    labels: _labels = {},
+    showExpiredLicenseBanner = false,
+    'data-test-subj': dataTestSubj,
+    size = 'm',
+  }) => {
+    // FIXME:PT should handle cases where we find the `id` to be invalid (show toast, close flyout ?)
+    // FIXME:PT should the flyout be made `reusable` for use cases where it gets opened in other areas of security (like Event filters)
+    // FIXME:PT handle cases where the "create" does not call the actual create api (ex. upload) (maybe defer from first iteration
+
+    const getTestId = useTestIdGenerator(dataTestSubj);
+    const isFlyoutOpened = useIsFlyoutOpened();
+    const setUrlParams = useSetUrlParams();
+    const { urlParams } = useUrlParams<{
+      id?: string;
+      show?: string;
+      [key: string]: string | undefined;
+    }>();
+
+    const labels = useMemo<typeof ARTIFACT_FLYOUT_LABELS>(() => {
+      return {
+        ...ARTIFACT_FLYOUT_LABELS,
+        ..._labels,
+      };
+    }, [_labels]);
+
+    const isEditFlow = urlParams.show === 'edit';
+
+    const handleFlyoutClose = useCallback(() => {
+      // FIXME:PT Question: Prevent closing it if update is underway?
```

Comment:
I would say yes too

Reply:
Cool. I'll go with that approach. I think it makes sense